src/xSTIR/cSTIR/stir_x.cpp

```diff
@@ -320,34 +320,34 @@ ListmodeToSinograms::estimate_randoms_()
 	ProjData& proj_data = *randoms_sptr->data();
 
 	const int num_rings =
-		template_projdata_ptr->get_proj_data_info_ptr()->get_scanner_ptr()->
+		template_projdata_ptr->get_proj_data_info_sptr()->get_scanner_ptr()->
 		get_num_rings();
 	const int num_detectors_per_ring =
-		template_projdata_ptr->get_proj_data_info_ptr()->get_scanner_ptr()->
+		template_projdata_ptr->get_proj_data_info_sptr()->get_scanner_ptr()->
 		get_num_detectors_per_ring();
 	DetectorEfficiencies& efficiencies = *det_eff_sptr;
 
 	{
-		const ProjDataInfoCylindricalNoArcCorr * const proj_data_info_ptr =
-			dynamic_cast<const ProjDataInfoCylindricalNoArcCorr * const>
-			(proj_data.get_proj_data_info_ptr());
-		if (proj_data_info_ptr == 0)
+		const shared_ptr<const ProjDataInfoCylindricalNoArcCorr> proj_data_info_sptr =
+			stir::dynamic_pointer_cast<const ProjDataInfoCylindricalNoArcCorr>
+			(proj_data.get_proj_data_info_sptr());
+		if (proj_data_info_sptr == 0)
 		{
 			error("Can only process not arc-corrected data\n");
 		}
 
 		const int mashing_factor =
-			proj_data_info_ptr->get_view_mashing_factor();
+			proj_data_info_sptr->get_view_mashing_factor();
 
 		shared_ptr<Scanner>
-			scanner_sptr(new Scanner(*proj_data_info_ptr->get_scanner_ptr()));
+			scanner_sptr(new Scanner(*proj_data_info_sptr->get_scanner_ptr()));
 		unique_ptr<ProjDataInfo> uncompressed_proj_data_info_uptr
 			(ProjDataInfo::construct_proj_data_info(scanner_sptr,
 				/*span=*/1, max_ring_diff_for_fansums,
 				/*num_views=*/num_detectors_per_ring / 2,
 				scanner_sptr->get_max_num_non_arccorrected_bins(),
 				/*arccorrection=*/false));
-		const ProjDataInfoCylindricalNoArcCorr * const uncompressed_proj_data_info_ptr =
+		const ProjDataInfoCylindricalNoArcCorr * const uncompressed_proj_data_info_sptr =
```

Comment:
This is the wrong name though. It isn't a `shared_ptr`. Introducing variables with the wrong name doesn't seem a great idea.

Can we revert this? Or change it to using a reference (probably clearest).

Reply:
Sorry, probably a case of "replace all" gone wrong. Ok, i'll change to ref and put in new PR.